### PR TITLE
konami/firebeat: Update pop'n music to use DVD CHDs + add new clones

### PR DESCRIPTION
**Pull request by 987123879113**
Rebuilt DVD CHDs for Firebeat pop'n music by using extractraw and then createdvd. The generated pop'n music 4 DVD CHD matches OG's original hash using this method.

Additionally, both popnanm and popnanm2 have a license period system presumably due to licensing agreements made with the anime studios in order to make these games. The dongles have the date they were made and the license period baked into the dongle, with the license period set to 5 years each. After 5 years when the license has expired, the games disable a bunch of graphics (basically anything that uses the logos or characters from the featured animes) and replace them with generic graphics.

Setting the period to 00/00 instead of 05/00 is enough to turn the dongles into "forever" license dongles, so I made the patch to the existing dongle dumps and marked them as BAD_DUMP because they're modified data. But probably a real forever license wouldn't be found in the wild.

The license expired versions have significant enough differences, even if it's a worse experience, to the fully featured versions so I added the expired license versions as clones and the forever license version as the parents.

Lastly, all Firebeat pop'n music games have been promoted to working since they're running on par with the rest of the games in the driver now that the DVD images can be loaded.

### CODE REVIEW

File: src/mame/konami/firebeat.cpp [hunk 1]
```diff
@@ -2660,15 +2700,17 @@ GAMEL( 2000, kbm2nd, 0,   firebeat_kbm, kbm, firebeat_kbm_state, init_kbm_jp, RO
 GAMEL( 2001, kbm3rd, 0,   firebeat_kbm, kbm, firebeat_kbm_state, init_kbm_jp, ROT270, "Konami", "Keyboardmania 3rd Mix", MACHINE_IMPERFECT_SOUND, layout_firebeat )
 
 // Requires DVD CHD support. Once DVD CHD support is implemented then MACHINE_NOT_WORKING can be removed
```
Comment (einstein95) [inline src/mame/konami/firebeat.cpp:2702]: Should this comment be updated now that it's been done?

Comment (987123879113) [reply src/mame/konami/firebeat.cpp]: Good catch, completely missed that. Fixed.

**Comment by etabeta78** [comment]
Rather than creating separate bad_dumps for the hacked dongles in the
clones, can’t you load the original one and add a ROM_FILL with the 00
value on the specific byte that needs to be changed?
This way we avoid spreading the hacked dongle around…

My 2c

**Comment by startaq** [comment]
Maybe instead of a clone set it could be done with a configuration switch? Something like "License expired: Yes/No".

**Comment by DavidHaywood** [comment]
I think clone sets are fine in this case.

If somebody were to want to repair a board to original state they'd know they had to create a cloned dongle with that data.  Config switch would be more of a hack.

**Comment by ICEknigh7** [comment]
Can't the dongle selection be done as with the BIOS, before the emulation starts? That would both avoid duplication for each CHD and allow for preservation of the original.

**Comment by 987123879113** [comment]
>This way we avoid spreading the hacked dongle around…

I originally considered using `ROM_FILL` just to avoid making a new dongle file for this but not wanting to spread a hacked dongle isn't a good reason not to make it a new ROM file. They're valid dongles and it's the only way the games can be played as they were when they were originally released, and the original real dongle dumps are also preserved in the clones so there's no real downsides here.

>Can't the dongle selection be done as with the BIOS, before the emulation starts? That would both avoid duplication for each CHD and allow for preservation of the original.

The two clones are parented to the main games so unless you're storing entirely different folders for every clone you have, you shouldn't have any duplication. Plus, using the BIOS setting is a gross hack and will have other knock on effects such as the dongles showing up as BIOS files in `-listxml`. Not really an option I'm willing to consider.

>Maybe instead of a clone set it could be done with a configuration switch? Something like "License expired: Yes/No".

This seems like the most feasible out of the suggestions but I'm still not really convinced it's necessary, and it relies on the code not being refactored. I didn't realize the Dallas iButton emulation code was all just hanging out in the Firebeat driver. It really should be rewritten into a proper separate device. So short term it would work but long term I don't think a configuration switch would be a good solution.

**Comment by ICEknigh7** [comment]
> using the BIOS setting is a gross hack and will have other knock on effects such as the dongles showing up as BIOS files in -listxml. Not really an option I'm willing to consider.

I didn't mean to use the BIOS setting, but rather something akin to what's done with them (and with software lists), i.e., ask which dongle to load before emulation starts.

**Comment by rb6502** [comment]
I'm on team clone sets.  I think given MAME ROMs exist to help people keep real boards alive, it's important to have sets that will make the machine work like it originally did.

**Comment by Gurudumps** [comment]
RB: Yeah a lot of people are making money hacking boards and don't want this secret info spread around. Sorry hack$$ guys, you're in the wrong place because MAME is all about preservation and making things public and open.

**Pull request merged**